src/features/aptosSignAndSubmitTransaction.ts

```diff
@@ -24,6 +25,12 @@ export type AptosSignAndSubmitTransactionMethod = (
   transaction: AptosSignAndSubmitTransactionInput
 ) => Promise<UserResponse<AptosSignAndSubmitTransactionOutput>>
 
-export type AptosSignAndSubmitTransactionInput = AnyRawTransaction
-/** Output of signing transactions. */
-export type AptosSignAndSubmitTransactionOutput = PendingTransactionResponse
+export interface AptosSignAndSubmitTransactionInput {
+  gasUnitPrice?: number;  // defaults to estimated gas unit price
+  maxGasAmount?: number;  // defaults to estimated max gas amount
+  payload: TransactionPayload | InputGenerateTransactionPayloadData;
```

Comment:
what is the reason for the different types...?

Reply:
`TransactionPayload` is the BCS-serializable class instance (you can build a `RawTransaction` with it)
`InputGenerateTransactionPayloadData` is the pojo format (cannot directly build a `RawTransaction` and have to fetch the ABI and convert first)

Reply:
gotit. what are the different use cases that we would need both on the standard? 

If looking at the dapp<>adapter<>wallet flow, the dapp sends the json object -> adapter builds the transaction payload (either TransactionPayload or InputGenerateTransactionPayloadData )?

Reply:
the dapp might want to send the payload instance too

Reply:
we can start with just the pojo payload if you prefer, works for me

Reply:
yes lets  do that, the adapter doesnt support the TransactionPayload type anyway